src/WebJobs.Script/Binding/BlobBinding.cs

```diff
@@ -38,14 +38,25 @@ public override bool HasBindingParameters
 
             boundBlobPath = Resolve(boundBlobPath);
 
-            Stream blobStream = context.Binder.Bind<Stream>(new BlobAttribute(boundBlobPath, FileAccess));
-            if (FileAccess == FileAccess.Write)
+            // TEMP: If the target type is set, use the binder logic to get a value of that type.
+            if (context.TargetType != null)
             {
-                await context.Value.CopyToAsync(blobStream);
+                context.Value = await context.Binder.BindAsync(context.TargetType, new BlobAttribute(boundBlobPath, FileAccess));
```

Comment:
Below this binding handles both input/output. It doesn't look like your new TargetType block does? It only sets context.Value (input binding).